share/functions/funced.fish

```diff
@@ -104,10 +110,12 @@ function funced --description 'Edit function definition'
                 set -l new_checksum (__funced_md5 "$tmpname")
                 if test "$new_checksum" = "$checksum"
                     echo (_ "Editor exited but the function was not modified")
+                    # Don't source or save an unmodified file.
+                    break
                 end
             end
 
-            if not source $tmpname
+            if not source <$tmpname
```

Comment:
This changes it so that `source` doesn't know about the filename anymore.

That means after you've `funced` a function once (without `--save`) it has no file anymore and was now defined on stdin.

Is there any particular reason for that? Would it not remember the tempfile, and then a second `funced` would use that again, so the things outside of the function stay there?

(still not perfect because what if you run `funced fish_git_prompt` and then `funced __fish_git_prompt_operation_branch_bare` or any other of the helper functions)

Reply:
The problem is that the temporary file is cleaned up by `funced`. So yes, without `--save` you may lose the ability to edit the file properly again - hence the warnings. We could leave the tempfile around, perhaps by moving it to `XDG_CACHE_HOME` (people do run programs like `tmpreaper`).